packages/api-headless-cms-ddb/src/operations/entry/index.ts

```diff
@@ -600,6 +600,80 @@ export const createEntriesStorageOperations = (
         }
     };
 
+    const restore: CmsEntryStorageOperations["restore"] = async (initialModel, params) => {
+        const { entry, storageEntry: initialStorageEntry } = params;
+        const model = getStorageOperationsModel(initialModel);
+
+        /**
+         * First we need to load all the revisions and published / latest entries.
+         */
+        const queryAllParams: QueryAllParams = {
+            entity,
+            partitionKey: createPartitionKey({
+                id: entry.id,
+                locale: model.locale,
+                tenant: model.tenant
+            }),
+            options: {
+                gte: " "
+            }
+        };
+
+        let records: DbItem<CmsEntry>[] = [];
+        try {
+            records = await queryAll(queryAllParams);
+        } catch (ex) {
+            throw new WebinyError(
+                ex.message || "Could not load all records.",
+                ex.code || "LOAD_ALL_RECORDS_ERROR",
+                {
+                    error: ex,
+                    id: entry.id
+                }
+            );
+        }
+
+        const storageEntry = convertToStorageEntry({
+            model,
+            storageEntry: initialStorageEntry
+        });
+
+        /**
+         * Then create the batch writes for the DynamoDB, with the updated data.
+         */
+        const items = records.map(record => {
+            return entity.putBatch({
+                ...record,
+                ...storageEntry
```

Comment:
Am I reading this correct?

We update every retrieved record (revision) with `storagEntry`'s properties?